amethyst_renderer/src/pass/shaded_util.rs

```diff
@@ -2,88 +2,79 @@ use std::mem;
 
 use amethyst_core::GlobalTransform;
 use amethyst_core::specs::prelude::{Join, ReadStorage};
-use gfx::traits::Pod;
+
+use glsl_layout::*;
 
 use cam::Camera;
 use light::Light;
 use pipe::{Effect, EffectBuilder};
 use resources::AmbientColor;
 use types::Encoder;
 
-fn pad(x: [f32; 3]) -> [f32; 4] {
-    [x[0], x[1], x[2], 1.0]
-}
 
-#[repr(C)]
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Uniform)]
 pub(crate) struct FragmentArgs {
-    point_light_count: i32,
-    directional_light_count: i32,
+    point_light_count: uint,
+    directional_light_count: uint,
 }
 
-#[repr(C)]
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Uniform)]
 pub(crate) struct PointLightPod {
-    position: [f32; 4],
-    color: [f32; 4],
-    intensity: f32,
-    _pad: [f32; 3],
+    position: vec3,
+    color: vec3,
+    pad: float,
+    intensity: float,
 }
 
-unsafe impl Pod for PointLightPod {}
-
-#[repr(C)]
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Uniform)]
 pub(crate) struct DirectionalLightPod {
-    color: [f32; 4],
-    direction: [f32; 4],
+    color: vec3,
+    direction: vec3,
 }
 
-unsafe impl Pod for DirectionalLightPod {}
-
 pub(crate) fn set_light_args(
     effect: &mut Effect,
     encoder: &mut Encoder,
     light: &ReadStorage<Light>,
     ambient: &AmbientColor,
     camera: Option<(&Camera, &GlobalTransform)>,
 ) {
-    let point_lights: Vec<PointLightPod> = light
+    let point_lights: Vec<_> = light
         .join()
         .filter_map(|light| {
             if let Light::Point(ref light) = *light {
                 Some(PointLightPod {
-                    position: pad(light.center.into()),
-                    color: pad(light.color.into()),
+                    position: light.center.into(),
+                    color: [100.0, 0.0, 0.0].into(),
```

Comment:
Just doing a really really quick review.

Why is the color hardcoded?

Reply:
Debugging stuff. Nice catch.